x/incentive/keeper/iter.go

```diff
@@ -0,0 +1,42 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/umee-network/umee/v4/util"
+	"github.com/umee-network/umee/v4/util/store"
+	"github.com/umee-network/umee/v4/x/incentive"
+)
+
+// GetAllIncentivePrograms returns all incentive programs
+// that have been passed by governance and have a particular status.
+// The status of an incentive program is either Upcoming, Ongoing, or Completed.
+func (k Keeper) GetAllIncentivePrograms(ctx sdk.Context, status incentive.ProgramStatus) []incentive.IncentiveProgram {
+	programs := []incentive.IncentiveProgram{}
+
+	var prefix []byte
+	switch status {
+	case incentive.ProgramStatusUpcoming:
+		prefix = keyPrefixUpcomingIncentiveProgram
+	case incentive.ProgramStatusOngoing:
+		prefix = keyPrefixOngoingIncentiveProgram
+	case incentive.ProgramStatusCompleted:
+		prefix = keyPrefixCompletedIncentiveProgram
+	default:
+		return []incentive.IncentiveProgram{}
+	}
+
+	iterator := func(_, val []byte) error {
+		var p incentive.IncentiveProgram
+		if err := p.Unmarshal(val); err != nil {
+			// improperly marshaled IncentiveProgram should never happen
+			return err
+		}
+
+		programs = append(programs, p)
+		return nil
+	}
```

Comment:
Added pagination back to the query.

As for why I want to keep the `util/store/iter.go` functions - they save us from potential gotchas (e.g. missing iterator.Close) and have an easier function signature to work with (function manipulates `key, value, prefix` without requiring `Valid()` and `Next()` plus checks the `error` automatically.)

Once this is merged, we can switch [the iterate function](https://github.com/umee-network/umee/blob/main/x/leverage/keeper/iter.go#L14) in the leverage module to use this utility as well. I avoided it in this PR because I'm staying inside incentive module.